homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -184,6 +197,29 @@ def get_router_title(conn: Connection) -> str:
 
         return self.async_create_entry(title=title, data=user_input)
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle SSDP initiated config flow."""
+        # Try to distinguish from other non-LTE Huawei router devices
+        if (
+            discovery_info.get(ATTR_UPNP_DEVICE_TYPE)
+            != "urn:schemas-upnp-org:device:InternetGatewayDevice:1"
```

Comment:
You should be able to add device type to the manifest https://developers.home-assistant.io/docs/en/creating_integration_manifest.html#ssdp

Reply:
The docs say that discovery is invoked if _any_ of the specified info is found. I read that to mean that if I added device type to the manifest, I'd need to check both manufacturer and device type matches in the code (because I would be handed all cases where either of them matches), whereas I now only have to check device type there and manufacturer is taken care of for me. So adding device info to the manifest would just add complexity for no gain that I can see.

Now, if the info match in manifest data would require **all** specified bits to match (instead of any), then I could move this to the manifest.

Reply:
Verified the "any" behavior, so I guess it's a matter of taste whether to check manufacturer or device type in manifest and the other in code. Leaving as is for now.

Reply:
oh, interesting. That feels weird 🤔 Wonder if we should update the way it works? I don't think that we have any integrations relying on the `any` behavior

Reply:
I agree `and` would be better. Added that and some other improvements in #28285